### ISSUE TITLE
Under MacOS Sierra beta, Godot engine 2.1 UI is quite broken (for me)

### ISSUE THREAD

**Issue by n1rvana** [godotengine/godot#6301]
**Operating system or device - Godot version:**
macOS 10.12 Beta Build 16A304a
Godot 2.1 release (v2.1.stable.official)

**Issue description** (what happened, and what was expected):
Opening a project produces an unusable UI.  It appears many UI elements are missing and the overall UI is super zoomed. 

**Steps to reproduce:**
-- Download the example code onto a machine with the above beta
-- Import the example code folder via the Godot UI 
-- Double-Click on an item in the project list (Glass bubbles for example)
-- Godot relaunches with a full screen window UI that is broken (image attached)
-- two finger mouse scrolling in the screen can make some of the background image (from venice) show a bit more ,but nothing works as expected and there are no controls, etc.

-- also the "quit" functions for Godot do not work.  The Godot menu doesn't work (on the mac menu bar) and the right click menu in the dock has a "quit" option equivilent to hitting command-q but that does not work.  Have to force quit.

**Link to minimal example project** (optional but very welcome):

**Additional Notes:**

I have not attempted to build 2.2 from scratch.  I am running macOS betas as they are released.  Sierra is expected to be released within a month from now.  I can run test builds and help debug, and can build from scratch if need be, just let me know. 
![godot-initial](https://cloud.githubusercontent.com/assets/1340910/18027017/884e4518-6c1d-11e6-9008-aa687f871e35.jpeg)

**Comment by n1rvana** [comment]
So, I also tried building the master branch from source to see what would happen.  Results are identical.

Here's the command output on the command line from running the build from Master:

```
/bin/ (m2) 🌀  ./godot.osx.tools.32 
arguments
0: ./godot.osx.tools.32
ERROR: create: Condition ' !create_func ' is true. returned: 0
   At: core/os/mutex.cpp:39.
nscreen count 2
ERROR: set_locale: Invalid Locale: en-US
   At: core/translation.cpp:520.
ERROR: set_locale: Invalid Locale: fa
   At: core/translation.cpp:453.
ERROR: set_locale: Invalid Locale: id
   At: core/translation.cpp:453.
WARNING: ProjectManager: Asset Library not available, as it requires SSL to work.
   At: tools/editor/project_manager.cpp:1339.
OPENING: /Users/eon/codex/Godot/Godot-Demos-2.1-20160729/2d/texscreen (::Users::eon::codex::Godot::Godot-Demos-2.1-20160729::2d::texscreen)
arguments
0: /Users/eon/codex/Godot/godot/bin/godot.osx.tools.32
1: -path
2: /Users/eon/codex/Godot/Godot-Demos-2.1-20160729/2d/texscreen
3: -editor
ERROR: create: Condition ' !create_func ' is true. returned: 0
   At: core/os/mutex.cpp:39.
ERROR: free_static: Condition ' !MemoryPoolStatic::get_singleton() ' is true.
   At: core/os/memory.cpp:58.
/bin/ (m2) 🌀  nscreen count 2
ERROR: set_locale: Invalid Locale: en-US
   At: core/translation.cpp:520.
ERROR: set_locale: Invalid Locale: fa
   At: core/translation.cpp:453.
ERROR: set_locale: Invalid Locale: id
   At: core/translation.cpp:453.
WARNING: EditorNode: Asset Library not available, as it requires SSL to work.
   At: tools/editor/editor_node.cpp:6467.

/bin/ (m2) 🌀  
```

**Comment by n1rvana** [comment]
I'd like to hear if others can repro this issue, or if anyone knows of a work around.

The Sierra betas are coming frequently, and very stable, and given Apple's announcement schedule, will likely go golden master within the next 45 days, possibly the next 20 days!

**Comment by vnen** [comment]
Seems like the same problem as #5688.

**Comment by akien-mga** [comment]
@n1rvana Can you confirm if it's the same issue as #5688?

**Comment by chadams** [comment]
I'm also getting this issue. However, if you turn off multiple monitors (or mirror your display) the UI renders properly on Mac (must restart)

**Comment by n1rvana** [comment]
I see this is targeted to be fixed in 2.2.  I reproduced it in 2.1.1. Confirmed that the "Open in low resolution" workaround works.  Also the "set displays to mirrored" workaround works.  This is under macOS 10.12.2, fwiw.

**Comment by akien-mga** [comment]
Closing as duplicate of #5688.
